core/src/engine/io/tls_wrapper.cpp

```diff
@@ -579,6 +579,22 @@ size_t TlsWrapper::SendAll(const void* buf, size_t len, Deadline deadline) {
                              deadline, "SendAll");
 }
 
+[[nodiscard]] size_t TlsWrapper::WriteAll(std::initializer_list<IoData> list, Deadline deadline) {
+  size_t total_len = 0;
+  for (const auto& io_data : list) {
+    total_len += io_data.len;
+  }
+  char* buf = new char[total_len];
```

Comment:
Memory allocation is CPU consuming... Make an array on stack and merge small chunks of data into it